### PR TITLE
BNGP-2880 & BNGP-2809 Combined: Main Credits Journey: Individual or Organisation & Metric Upload

### PR DESCRIPTION
**Pull request by usman-j-shaikh**

### CODE REVIEW

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
Use of constants.landownerTypes needs to be replaced with use of costants.individualOrOrganisationTypes following he merge of https://github.com/DEFRA/biodiversity-net-gain-service/pull/515 to master.

**Review by pwadmore-ea — COMMENTED** [review]
I'm happy with the changes made so far.

Approval relies on alignment with the copydeck (URLs should be finalised but error messages are still to be confirmed).

File: packages/webapp/src/routes/credits-purchase/credits-metric-upload.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      return h.view(creditsPurchaseConstants.views.CREDITS_PURCHASE_UPLOAD_METRIC, {
+        err: [{
+          text: 'The selected file is empty',
+          href: UPLOAD_CREDIT_METRIC_ID
+        }]
+      })
+    case creditsPurchaseConstants.uploadErrors.noFile:
+      return h.view(creditsPurchaseConstants.views.CREDITS_PURCHASE_UPLOAD_METRIC, {
+        err: [{
+          text: 'Select a Biodiversity Metric',
```
Comment (paulsimonandrews) [inline packages/webapp/src/routes/credits-purchase/credits-metric-upload.js:38]: Error message should be `Upload the statutory biodiversity metric file with the ‘unit shortfall summary’ tab` to match the copydeck

File: packages/webapp/src/views/credits-purchase/credits-metric-upload-check.html
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-metric-upload-check.html:1]: This should be `/credits-purchase/check-metric-file` to match the copy deck

File: packages/webapp/src/views/credits-purchase/credits-metric-upload-check.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{% block formContent %}
+    <div class="govuk-body">
+        {% if err %}
+            {% include "errorbanner.html" %}
+        {% endif %}
+        {% set idPrefix = 'checkUploadMetric' %}
+        {% set name = 'checkUploadMetric' %}
+        {% set id = 'check-upload-correct-yes' %}
+        {% set downloadUrl = 'credits-download-metric' %}
+        {% set checkUploadCorrectYesText = 'Yes, this is the Secretary of State Metric file my local planning authority reviewed' %}
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-metric-upload-check.html:15]: Should be `Yes, this is the statutory biodiversity metric file my local planning authority reviewed` to match the copy deck

File: packages/webapp/src/routes/credits-purchase/credits-metric-upload-check.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      return h.redirect(creditsPurchaseConstants.routes.CREDITS_PURCHASE_UPLOAD_METRIC)
+    } else if (checkUploadMetric === creditsPurchaseConstants.options.creditsCheckUploadMetric.YES) {
+      return h.redirect(creditsPurchaseConstants.routes.CREDITS_PURCHASE_CONFIRM_DEV_DETAILS)
+    }
+    return h.view(creditsPurchaseConstants.views.CREDITS_PURCHASE_CHECK_UPLOAD_METRIC, {
+      filename: path.basename(metricUploadLocation),
+      ...getContext(request),
+      err: [
+        {
+          text: 'Select yes if this is the correct file',
```
Comment (paulsimonandrews) [inline packages/webapp/src/routes/credits-purchase/credits-metric-upload-check.js:28]: Should be `Select yes if this is the statutory biodiversity metric file your local planning authority reviewed with your biodiversity net gain statement` to match the copy deck

File: packages/webapp/src/routes/credits-purchase/credits-development-location.js
Comment (paulsimonandrews) [inline packages/webapp/src/routes/credits-purchase/credits-development-location.js:1]: Should be `credits-purchase/confirm-development-details` to match the copy deck

File: packages/webapp/src/views/credits-purchase/credits-development-location.html
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-development-location.html:1]: Should be `credits-purchase/confirm-development-details` to match the copy deck

File: packages/webapp/src/views/credits-purchase/credits-development-location.html [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    {% set id = 'dev-details-checked-yes' %}
+  </div>
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <h1 class="govuk-heading-xl">
+        {{ pageHeading }}
+      </h1>
+    </div>
+    <div class="govuk-grid-column-two-thirds">
+      <p>These details have been taken from the Secretary of State Metric file your local planning authority reviewed.</p>
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-development-location.html:22]: Should be `These details have been taken from the statutory biodiversity metric calculation tool your local planning authority reviewed.` to match the copy deck

File: packages/webapp/src/routes/credits-purchase/credits-development-location.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      return h.redirect(creditsPurchaseConstants.routes.CREDITS_PURCHASE_UPLOAD_METRIC)
+    } else if (confirmDevDetails === creditsPurchaseConstants.options.creditsCheckDetails.YES) {
+      return h.redirect(creditsPurchaseConstants.routes.CREDITS_PURCHASE_TASKLIST)
+    } else {
+      return h.view(creditsPurchaseConstants.views.CREDITS_PURCHASE_CONFIRM_DEV_DETAILS, {
+        filename: path.basename(metricUploadLocation),
+        ...await getContext(request),
+        err: [
+          {
+            text: 'Select yes if this is the correct file',
```
Comment (paulsimonandrews) [inline packages/webapp/src/routes/credits-purchase/credits-development-location.js:27]: Should be `Select yes if these details are correct` to match the copy deck

File: packages/webapp/src/routes/credits-purchase/applying-individual-organisation.js
Comment (paulsimonandrews) [inline packages/webapp/src/routes/credits-purchase/applying-individual-organisation.js:1]: Should be `/credits-purchase/purchasing-individual-organisation` to match the copydeck

File: packages/webapp/src/views/credits-purchase/applying-individual-organisation.html
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/applying-individual-organisation.html:1]: Should be `/credits-purchase/purchasing-individual-organisation` to match the copydeck

File: packages/webapp/src/routes/credits-purchase/applying-individual-organisation.js [hunk 6]
```diff
@@ -1,12 +1,64 @@
...
-    },
-    handler: (_request, h) => h.view(creditsPurchaseConstants.views.ESTIMATOR_CREDITS_INDIVIDUAL_ORG)
-  }
-]
+import constants from '../../utils/constants.js'
+import getOrganisationDetails from '../../utils/get-organisation-details.js'
+
+const individualSignInErrorMessage = 'Select \'individual\' if you are purchasing statutory biodiversity credits as an individual'
+
+const organisationSignInErrorMessage = 'Select \'organisation\' if you are purchasing statutory biodiversity credits as an organisation'
```
[the remark this answers is not in the document]
Comment (paulsimonandrews) [reply packages/webapp/src/routes/credits-purchase/applying-individual-organisation.js:6]: These are the current messages in the latest version of the copydecks so I suggest we leave them and ask design to review the build for changes. We shouldn't be making these decisions!

[the remark this answers is not in the document]
Comment (paulsimonandrews) [reply packages/webapp/src/routes/credits-purchase/applying-individual-organisation.js:6]: So I suggest leave these as they are for now.

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Just content changes to match the latest versions of the copydecks.

Please update and rename routes, views and test files to follow the latest URLs

**Review by paulsimonandrews — APPROVED** [review]

**Comment by shaundefra** [comment]
tested locally

**Pull request merged**